theano/gpuarray/opt.py

```diff
@@ -697,6 +698,28 @@ def local_gpua_elemwise(op, context_name, inputs, outputs):
         name = 'Gpu' + name
     if len(outputs) > 1:
         return
+    have_cuda = False
+    have_opencl = False
+    if inputs and isinstance(inputs[0].type, GpuArrayType):
```

Comment:
No, I have not seen such cases! The condition is here just for precaution, as I access inputs[0] attributes after.